### PR TITLE
SDK-247 : Add PHP support for AML service

### PR DESCRIPTION
**Pull request by sidibos**

### CODE REVIEW

File: src/Yoti/YotiClient.php [hunk 1]
```diff
@@ -200,103 +184,195 @@ public function getActivityDetails($encryptedConnectToken = null)
...
+            $signedRequest->getApiRequestUrl($this->_connectApi),
+            $amlPayload,
+            RestRequest::METHOD_POST
+        );
+
+        $result = $restRequest->exec();
+
+        // Get response data array
+        $responseArr = json_decode($result['response'], TRUE);
+        // Check if there is a json decode error
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: JSON

File: src/Yoti/YotiClient.php [hunk 2]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
         return $attributeList;
     }
 
+    /**
+     * Validate and return pem content.
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: PEM

File: src/Yoti/YotiClient.php [hunk 3]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
...
+     *
+     * @param string bool|$pem
+     *
+     * @throws \Exception
+     */
+    public function processPem(&$pem)
+    {
+        // Check pem passed
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: PEM

File: src/Yoti/YotiClient.php [hunk 4]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
...
+        if(!$pem)
+        {
+            throw new \Exception('PEM file is required', 400);
+        }
+
+        // Check that file exists if user passed pem as a local file path
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: PEM

File: src/Yoti/YotiClient.php [hunk 5]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
...
+            throw new \Exception('PEM file was not found.', 400);
+        }
+
+        // If file exists grab the content
+        if(file_exists($pem))
+        {
+            $pem = file_get_contents($pem);
+        }
+
+        // Check key is valid
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: Check if

File: src/Yoti/YotiClient.php [hunk 6]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
...
+        if(!openssl_get_privatekey($pem))
+        {
+            throw new \Exception('PEM key is invalid', 400);
+        }
+    }
+
+    /**
+     * Check Sdk Id is provided.
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: SDK ID

File: src/Yoti/YotiClient.php [hunk 7]
```diff
@@ -428,12 +489,80 @@ private function getAttributeList(EncryptedData $encryptedData, $wrappedReceiptK
...
+     *
+     * @param string $sdkId
+     *
+     * @throws \Exception
+     */
+    public function checkSdkId($sdkId)
+    {
+        // Check sdk id passed
```
Comment (vassyz) [inline src/Yoti/YotiClient.php]: SDK ID

File: src/Yoti/Http/AbstractRequest.php [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    /**
+     * Check if the method can include payload data.
+     *
+     * @param string $httpMethod
+     *
+     * @return bool
+     */
+    public static function canSendPayload($httpMethod)
+    {
+        $methodsThatIncludePayload = explode(',', self::METHODS_THAT_INCLUDE_PAYLOAD);
```
[the remark this answers is not in the document]
Comment (sidibos) [reply src/Yoti/Http/AbstractRequest.php]: Yes, but I can't use `$this` inside a static function and don't want to use `constant array syntax` which will requires `PHP 5.6`.

File: src/Yoti/Entity/Country.php [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    public function __construct($code, $name = NULL)
+    {
+        $this->code = $code;
+        $this->name = $name;
+    }
+
+    /**
+     * @param $name
+     */
+    public function setName($name)
```
[the remark this answers is not in the document]
Comment (sidibos) [reply src/Yoti/Entity/Country.php]: Just in case the name is provided as well, but I can remove the name part.

**Comment by sidibos** [comment]
@qzagarese it's been already tested against staging endpoint and it worked fine.

File: README.md [hunk 10]
```diff
@@ -166,6 +169,58 @@ The set of attributes the user has configured for the transaction.
...
+
+Yoti provides an AML (Anti Money Laundering) check service to allow a deeper KYC process to prevent fraud. This is a chargeable service, so please contact [sdksupport@yoti.com](mailto:sdksupport@yoti.com) for more information.
+
+Yoti will provide a boolean result on the following checks:
+* PEP list - Verify against Politically Exposed Persons list
+* Fraud list - Verify against  US Social Security Administration Fraud (SSN Fraud) list
+* Watch list - Verify against watch lists from the Office of Foreign Assets Control
+
+To use this functionality you must ensure:
+* Your application is assigned to your Organisation in the Yoti Dashboard - please see [here]('https://www.yoti.com/developers/documentation') for further information.
```
Comment (vassyz) [inline README.md]: Quotes should be removed `[here](https://www.yoti.com/developers/documentation)`

**Pull request merged**